File: 0100-same-tree/0100-same-tree.java
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        return check(p, q);
    }

    private boolean check(TreeNode a, TreeNode b) {
        if (a == null && b == null)
            return true;
        if(a==null || b==null) return false;
        else
            return (a.val == b.val) && check(a.left, b.left) && check(a.right, b.right);
    }
}

File: 0100-same-tree/0100-same-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool check(struct TreeNode* a,struct TreeNode* b){
    if(a==NULL && b==NULL) return true;
    if(a==NULL||b==NULL) return false;
    else
    return (a->val==b->val) && check(a->left,b->left) && check(a->right,b->right);
}
bool isSameTree(struct TreeNode* p, struct TreeNode* q) {
    return check(p,q);
}